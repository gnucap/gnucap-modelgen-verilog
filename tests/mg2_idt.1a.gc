attach ./modelgen_0.so
attach ../vams/vsine.so

verilog

`modelgen
module test_idt1a(p, n, cp, cn);
	electrical p, n, cp, cn;
	(* desc="" *) real idt1a;
	(* desc="" *) real idt2;
	analog begin
		V(p,n) <+ idt(V(cp, cn));
	end
endmodule

!make test_idt1a.so > /dev/null
attach ./test_idt1a.so

verilog
parameter r=1
parameter v=0
parameter pi=3.1415926535897932384626433832795028841972

test_idt1a #() dut(3,0,1,0);
vsine #(.freq(1/6.28), .dc(v), .ampl(1)) v1(1, 0);

list

print tran v nodes idt1(dut) idt2(dut) v(dut.*)
print tran + iter(0)
tran 0 {2*pi} {.2*pi} trace=n
status notime
end
