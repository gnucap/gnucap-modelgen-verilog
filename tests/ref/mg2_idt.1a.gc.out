parameter r=1 
parameter v=0 
parameter pi=3.1415926535897932384626433832795028841972 
test_idt1a #() dut (.p(3),.n(0),.cp(1),.cn(0));
vsine #(.freq(1/6.28),.ampl(1),.dc(v)) v1 (.p(1),.n(0));
#Time       v(1)       v(3)       idt1(dut)  idt2(dut)  v(dut._b_p_n) v(dut._b_idt_0_p_idt_0_n) iter(0)   
 0.         0.         0.         ??         0.         0.         0.         3.        
 0.62832    0.58804    0.19044    ??         0.         0.19044    0.19044    42.       
 1.2566     0.95125    0.68648    ??         0.         0.68648    0.68648    6.        
 1.885      0.95076    1.2994     ??         0.         1.2994     1.2994     6.        
 2.5133     0.58675    1.7949     ??         0.         1.7949     1.7949     6.        
 3.1416    -0.0015935  1.9834     ??         0.         1.9834     1.9834     6.        
 3.7699    -0.58933    1.7978     ??         0.         1.7978     1.7978     3.        
 4.3982    -0.95174    1.3137     ??         0.         1.3137     1.3137     3.        
 5.0265    -0.95027    0.71612    ??         0.         0.71612    0.71612    3.        
 5.6549    -0.58546    0.23366    ??         0.         0.23366    0.23366    3.        
 6.2832     0.0031869  0.046019   ??         0.         0.046019   0.046019   6.        
Gnucap   System status
iterations: op=0, dc=0, tran=87, fourier=0, total=116
transient timesteps: accepted=24, rejected=5, total=29
nodes: user=2, subckt=0, model=2, total=4
dctran density=75.0%, ac density=75.0%
